### PR TITLE
fix(build-ios): Use BUILD_DIR instead of CONFIGURATION_BUILD_DIR

### PR DESCRIPTION
**Pull request by mbektchiev**
Passing `CONFIGURATION_BUILD_DIR` to `xcodebuild` causes
build errors with projects having `nativescript-plugin-firebase`
when using Xcode 10's new build system similar to the following:
```
Build system information
error: Multiple commands produce '<PbxCp <project-path>/platforms/ios/build/src/core/ext/filters/client_channel/backup_poller.h>':
1) Target 'gRPC-Core' (project 'Pods') has copy command from '<project-path>/platforms/ios/Pods/gRPC-Core/src/core/ext/filters/client_channel/backup_poller.h' to '<project-path>/platforms/ios/build/src/core/ext/filters/client_channel/backup_poller.h'
2) Target 'gRPC-C++' (project 'Pods') has copy command from '<project-path>/platforms/ios/Pods/gRPC-C++/src/core/ext/filters/client_channel/backup_poller.h' to '<project-path>/platforms/ios/build/src/core/ext/filters/client_channel/backup_poller.h'
```

## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [x] Tests for the changes are included.

related to #4311

### CODE REVIEW

File: lib/services/ios-project-service.ts [hunk 1]
```diff
@@ -407,8 +416,8 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 		args = args.concat((buildConfig && buildConfig.architectures) || this.getBuildArchitectures(projectData, buildConfig, ["armv7", "arm64"]));
 
 		args = args.concat([
-			"-sdk", "iphoneos",
-			"CONFIGURATION_BUILD_DIR=" + path.join(projectRoot, "build", "device")
+			"-sdk", DevicePlatformSdkName,
+			"BUILD_DIR=" + path.join(projectRoot, "build")
```
Comment (Fatme) [inline lib/services/ios-project-service.ts]: `path.join(projectRoot, constants.BUILD_DIR)`

File: lib/services/ios-project-service.ts [hunk 2]
```diff
@@ -574,10 +583,10 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 			.concat(architectures)
 			.concat([
 				"build",
-				"-configuration", buildConfig.release ? "Release" : "Debug",
-				"-sdk", "iphonesimulator",
+				"-configuration", getConfigurationName(buildConfig.release),
+				"-sdk", SimulatorPlatformSdkName,
 				"ONLY_ACTIVE_ARCH=NO",
-				"CONFIGURATION_BUILD_DIR=" + path.join(projectRoot, "build", "emulator"),
+				"BUILD_DIR=" + path.join(projectRoot, "build"),
```
Comment (Fatme) [inline lib/services/ios-project-service.ts]: `path.join(projectRoot, constants.BUILD_DIR)`

**Comment by Fatme** [comment]
run ci

**Comment by Fatme** [comment]
We need to verify the `publish` command works as expected.

**Comment by rosen-vladimirov** [comment]
run ci

**Pull request merged**